controllers/azuremachine_controller.go

```diff
@@ -149,6 +150,20 @@ func (r *AzureMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, ret
 
 	// Always close the scope when exiting this function so we can persist any AzureMachine changes.
 	defer func() {
+		if machineScope.IsControlPlane() {
```

Comment:
is this if statement necessary, given that if and else branch apparently are the same?

Reply:
will remove it